sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/StandardRangeShardingAlgorithm.java

```diff
@@ -0,0 +1,96 @@
+import org.apache.shardingsphere.api.sharding.standard.PreciseShardingValue;
+import org.apache.shardingsphere.api.sharding.standard.RangeShardingValue;
+
+import java.math.RoundingMode;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Standard range sharding algorithm.
+ * <p>
+ * Standard range sharding algorithm is similar to the rule of partition table, but it can only be split by the same size.
+ * User can specify the range by setting `partition.lower`, `partition.upper` and `partition.volume` parameters.
+ * The `partition.volume` parameter determines the size of each partition.
+ * </p>
+ * <p>
+ * For example: If the `partition.lower` parameter is set to `10`, the `partition.upper` parameter is set to `45`,
+ * and the `partition.volume` parameter is set to `10`. The values in range [10,45] will be split to different partitions
+ * ——[10,20), [20, 30), [30, 40), [40, 45), and other values will be split to (-∞, 10) and [45, +∞).
+ * </p>
+ */
+public final class StandardRangeShardingAlgorithm extends AbstractRangeShardingAlgorithm {
+
+    private static final String PARTITION_LOWER = "partition.lower";
+
+    private static final String PARTITION_UPPER = "partition.upper";
+
+    private static final String PARTITION_VOLUME = "partition.volume";
+
+    private Map<Integer, Range<Long>> partitionRangeMap;
+
+    @Getter
+    @Setter
+    private Properties properties = new Properties();
+
+    @Override
+    public String doSharding(final Collection<String> availableTargetNames, final PreciseShardingValue<Long> shardingValue) {
+        checkInit();
+        return getTargetNameByPreciseShardingValue(availableTargetNames, shardingValue, partitionRangeMap);
+    }
+
+    @Override
+    public Collection<String> doSharding(final Collection<String> availableTargetNames, final RangeShardingValue<Long> shardingValue) {
+        checkInit();
+        return getTargetNameByRangeShardingValue(availableTargetNames, shardingValue, partitionRangeMap);
+    }
+
+    @Override
+    public String getType() {
+        return "STANDARD_RANGE";
+    }
+
+    @Override
+    public void initProperties() {
+        Preconditions.checkNotNull(properties.get(PARTITION_LOWER), "Standard range sharding algorithm partition lower cannot be null.");
+        Preconditions.checkNotNull(properties.get(PARTITION_UPPER), "Standard range sharding algorithm partition upper cannot be null.");
+        Preconditions.checkNotNull(properties.get(PARTITION_VOLUME), "Standard range sharding algorithm partition volume cannot be null.");
+        long lower = Long.parseLong(properties.get(PARTITION_LOWER).toString());
+        long upper = Long.parseLong(properties.get(PARTITION_UPPER).toString());
+        long volume = Long.parseLong(properties.get(PARTITION_VOLUME).toString());
+        Preconditions.checkArgument(upper - lower >= volume, "Standard range sharding algorithm partition range can not be smaller than volume.");
+
```

Comment:
A redundant blank line is expected to remove.

Reply:
> A redundant blank line is expected to remove.

Ok, I will modify it.